src/applications/disability-benefits/686c-674/README.md

```diff
@@ -0,0 +1,71 @@
+
+As mentioned we broke each individual chapter of the form into it's own folder and put each page from each of the form into it's own folder inside the chapter folder as well. We then have an `index.js` file inside the chapter folder that imports all of the pages and then exports them deconstructed so that they are easier to import and use inside `/config/form.js`. Inside each of the page folders there is a jsx file named for the page as well as any helper files we used to accomplish that page. Inside the JSX file we export a `schema` object as well as a `uiSchema` object, these are what is used inside `/config/form.js` to build the form.
+
+## Callouts on how the front end works
+
+We used the original form system to build this application and we followed a pretty standard implementation with just a few areas worth noting.
+
+### Valid VA file number check (front end)
+
+The form is built using `/config/form.js` which is imported and used inside `/containers/App.jsx` with the `<RoutedSavableApp />` component from the platform. In `/containers/App.jsx` we dynamically render a loading indicator if the app is in an `isLoading` state from Redux. This allows us to show a loading indicator while we make the call to check if the Veteran has a valid VA file number as noted in the block quote above. The back end of that valid VA file number is detailed below in the back end section. Once the call is made and the form renders we either show the user an alert if they do not have a valid VA file number, or a `<SaveInProgressIntro />` component from the platform to start the form.
+
+### The wizard
+The first page a Veteran sees when they enter the form from the introduction page is a "wizard" where they can choose what task they want to path they want to take through the form. The original design for our form intended for us only to show the chapters that the Veteran needed to see; the idea would be that we would show the Veteran a list of tasks they could accomplish using the form and they could pick whatever ones they wanted to perform, then we would only show the Veteran the chapters they needed to see to accomplish those tasks.
+
+The way we do this in the code is by using the `depends` attribute for the optional chapters inside `/config/form.js` along with a helper method called `isChapterFieldRequired` and passing in the `formData` object and the name of that respective chapter. The `isChapterFieldRequired` function then checks if the Veteran selected the task in the wizard associated with that chapter, which we hold in the `formData["view:selectable686Options"]` array in the form data and returns `true` and shows that chapter if it corresponds to a task the Veteran selected.
```

Comment:
It might be worth linking to the location of `isChapterFieldRequired` so the reader can take a look at the function directly.

Reply:
Yeah I think you are right. I was a little concerned with linking directly to anything within the codebase in case stuff moves around but I don't think this will move much.